src/frequenz/sdk/actor/channel_registry.py

```diff
@@ -0,0 +1,52 @@
+"""A class that would dynamically create, own and provide access to channels.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+
+from typing import Any, Dict, Generic, TypeVar
+
+from frequenz.channels import Broadcast, Receiver, Sender
+
+T = TypeVar("T")
+
+
+class ChannelRegistry(Generic[T]):
```

Comment:
Is Generic[T] needed? We would like channels to be identified by `str`, yes?

Reply:
Not necessarily. A channel can identified by a pair `ComponentId` and `MetricType`. But we don't know all use cases of the Registry, so let's leave it generic.

Reply:
I had the same question. IMHO this adds a bit of unnecessary complexity (and will force us to create different registries for different key types). I think we can require users to just convert to a string themselves (`f"{component_id}_{metric_typ}"`).